### PR TITLE
Port /tg/ crew pinpointer

### PR DESCRIPTION
**Pull request by Tayyyyyyy**
Quality of life thing for medical personnel who have to go look for people, no more constantly refreshing your crew monitor.

Gave one to the paramedic, one to the brig phys, and an extra to the CMO.

Only works on people with suit sensors set to maximum.

It's blue to distinguish it from the other pinpointers.

edit: oops, I "refactored" the normal pinpointer. Turns out that was a bad idea. Fixing...
edit: fixed!

person_to_track

pinpointer_sprite

:cl: Tayyyyyyy
add: Paramedic, Blueshield, and Medivends get crew pinpointers
/:cl:

### CODE REVIEW

**Comment by Aurorablade** [comment]
is this relying on suit sensor state?

**Comment by taukausanake** [comment]
> Only works on people with suit sensors set to maximum

Unless he edited it in afterwards.
I don't know, this seems a little OP for assassin antags. To know exactly where your target will be. Doesn't matter too much if suit sensors are on high, you're going to keep them on either way

**Comment by Tayyyyyyy** [comment]
Yep! It's just like the /tg/ one. your sensors have to have coordinates on

**Comment by Tayyyyyyy** [comment]
@taukausanake assassins already have the option of the advanced pinpointer. And this doesn't give that much more information than the crew monitor. It just refreshes way faster since the crew monitor code is so slow.

**Comment by Ralta** [comment]
Love, love, LOVE this idea. It's no more _powerful_ than the Crew Monitor, just more usable / less frustrating. 

I imagine most Blueshields are gonna ask the CMO for theirs, though...

**Comment by Spartan6** [comment]
Does it use alphabetical order? The screenshot implies it doesn’t.

**Comment by Tayyyyyyy** [comment]
> I imagine most Blueshields are gonna ask the CMO for theirs, though...

I think it would be pretty useful for the blueshield! But they're already carrying the normal pinpointer usually so it might be a little weird to have so many pinpointers?

> Does it use alphabetical order? The screenshot implies it doesn’t.

It doesn't!

**Comment by Spacemanspark** [comment]
We intentionally made the crew monitoring console take longer to refresh at one point, because we didn't want people tracking others too easily. Not sure how I feel about this.

**Comment by Tayyyyyyy** [comment]
> We intentionally made the crew monitoring console take longer to refresh at one point, because we didn't want people tracking others too easily. Not sure how I feel about this.

Just to be clear, the refresh rate on this can be adjusted, too. The problem with the handheld crew monitor is that here's the process for tracking someone:

- Open crew monitor (which takes some time for the html to render, can occasionally crash/lag your client if you do this too fast)
- Switch to crew monitor window
- Find yourself (on highpop, this can require scrolling)
- Find the other person (more scrolling)
- mentally calculate the direction you have to go in

This is a terrible experience. If we want it to be harder to track people, on this pinpointer, you can simply adjust the refresh rate.

Also, for perspective, on /tg/ this is provided in the medical vendors instead of the handheld crew monitors. So it's even more commonplace than in this PR.

**Comment by Shadeykins** [comment]
@Tayyyyyyy Playing devils advocate here, but why do you need to find yourself?

Finding your target is as easy as opening it up, scrolling an alphabetical list, and seeing the location appended next to their name.

I'm not sure how I feel about this, the CMO doesn't really have a use for it and it will end up in the hands of the Blueshield, and the Brig Physician shouldn't really be trawling maintenance for dead security officers.

It would be a nice tool for the paramedic, and the paramedic alone, to have.

**Comment by Ralta** [comment]
@Shadeykins I feel that although they won't be trawling maintenance, they can at least pinpoint injured personnel FAR more easily. 

Brig physicians often run to sites where crew and officers are injured during big fights or after a rescue, and it would help to at least point rescue crews in the right direction.

As for the CMO, they'd have it to use or give away at their discretion; I think having 3 on station total is pretty reasonable, as opposed to adding them to RnD or something.

**Comment by Tayyyyyyy** [comment]
@Shadeykins Maybe I play too much pod pilot...sometimes the location on the crew monitor is something like Space (150, 231) or Science Maintenance (174, 80) or something, and at that point you can't necessarily go off of just your target person.

I agree with Ralta! I honestly don't believe these are overly powerful. Even if blueshield manages to steal one from the CMO, it's not that effective because most antags will turn off suit sensors already. At worst, this is a slight buff to CMO/paramedic antags, but even then, it probably doesn't increase their power by that much.

**Comment by imsxz** [comment]
Its just a QOL thing, it's no more powerful than the crew monitor for tracking people, unless the crew monitor was intentionally designed to be clunky as fuck to use. I don't recognize the balance concerns presented by others here.

Having played with it on TG, it's really nice for doctors and antagonists to have, and I've never seen any complaints about it being too OP there for as long as I've been playing.

**Comment by Kyep** [comment]
Paramedic? Sure. Finding injured crew is his job.
CMO? I don't see why they need one. They should generally not be leaving medbay. 
Brig Phys? Again, isn't it medical's job to rescue downed crew? Giving a 'locate any crew member with sensors maxed' device to a member of security seems... prone to misuse. 
If I were going to give this to anyone, other than the Paramedic, it would be the Blueshield. Yes, I recognize they have one pinpointer already, but I see zero reason why that's relevant.

**Comment by Tayyyyyyy** [comment]
The CMO has an extra to give at their discretion, just like they have a bunch of other fancy equipment that is usually distributed to areas of medbay that need it. /tg/ has these in the medivends. I thought doing that would be too drastic, so I just gave one to the CMO. Maybe I should just leave one in the medical equipment locker?

Brig phys needs to help wounded officers. This will impact balance the least, because the brig phys is mindshielded. You can't use this to track down antags (their suit sensors are off) or a victim an antag is trying to hide (suit sensors are off). There is very very little room for abuse here. It's intended for brig physicians who need to find a wounded officer out in the field and patch them up.

I'm not opposed to giving it to the blueshield. I'll add it in if nobody is opposed to this idea.

**Comment by Spartan6** [comment]
If you’re giving one to the blueshield, remove the one for the CMO.

**Comment by Ralta** [comment]
I would rather it was given to the CMO instead of the Blueshield, but I agree that if it's going to the Blueshield, the CMO should lose theirs.

**Comment by Tayyyyyyy** [comment]
All right, in the interest of making sure CMO doesn't run into maints and get brutally murdered/thralled, I've removed their pinpointer.

**Comment by Ralta** [comment]
D'aww, fair enough. I guess that makes sense. Saying it like that, I can actually imagine CMO's trying to justify going into maint because "Why would we have been given a pinpointer if we weren't supposed to go to maint!?" 😛

**Comment by Fox-McCloud** [comment]
If this is going to be a thing, I have no real interest in it being locked behind the paramedic or brig phys's job (the latter of which makes ZERO sense to have, since their job is *supposed* to be healing prisoners and ***not*** officers).

Add these to the medical vendors instead.

**Comment by Tayyyyyyy** [comment]
Okay, I'll make those changes as soon as I can.

**Comment by SamHPurp** [comment]
I entirely disagree @Fox-McCloud - the Paramedic could use a nice bonus, and this could be it. Its a rather strong item, and loads of people having one could be quite a pain in the ass. It limited to the Paramedic makes a lot of sense to me.

File: code/game/gamemodes/nuclear/pinpointer.dm [hunk 1]
```diff
@@ -300,3 +310,83 @@
...
+	if(active)
+		active = FALSE
+		icon_state = icon_off
+		user.visible_message("<span class='notice'>[user] deactivates their pinpointer.</span>", "<span class='notice'>You deactivate your pinpointer.</span>")
+		return
+
+	var/list/name_counts = list()
+	var/list/names = list()
+
+	for(var/mob/living/carbon/human/H in mob_list)
```
Comment (SamHPurp) [inline code/game/gamemodes/nuclear/pinpointer.dm]: Is this going to respect Z-Levels, or are you going to get it pointing at the corresponding place in Wild West, when they're on the station?

Comment (Tayyyyyyy) [reply code/game/gamemodes/nuclear/pinpointer.dm]: In the trackable proc, it checks that they're on the same z-level. Is Wild West a gateway level?

Comment (SamHPurp) [reply code/game/gamemodes/nuclear/pinpointer.dm]: Thats good.

(And yes)

**Comment by tigercat2000** [comment]
Note: Description is incorrect, but changelog is correct. 
This is currently added to the
 - NanoMed Plus
 - Paramedic's locker
 - Blueshield's locker

@Tayyyyyyy Please update the description when you have a moment.
👍

**Pull request merged**